### PR TITLE
Allow progress colors to be settable and gettable from the internal host

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
If you look at the first commit, you'll see that it allows us to inherit the setting and getting of the Progress colors from the internal host.

The second+ commits are purely cosmetic since I was touching the file.

### CODE REVIEW

File: src/PowerShellEditorServices/Services/PowerShellContext/Session/Host/TerminalPSHostUserInterface.cs [hunk 1]
```diff
@@ -21,8 +21,9 @@ internal class TerminalPSHostUserInterface : EditorServicesPSHostUserInterface
     {
         #region Private Fields
 
-        private readonly PSHostUserInterface internalHostUI;
-        private ConsoleReadLine consoleReadLine;
+        private readonly PSHostUserInterface _internalHostUI;
+        private readonly PSObject _privateData;
+        private ConsoleReadLine _consoleReadLine;
```
Comment (TylerLeonhardt) [inline src/PowerShellEditorServices/Services/PowerShellContext/Session/Host/TerminalPSHostUserInterface.cs:26]: Codacy Issue found: [Make '_consoleReadLine' 'readonly'.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5428870)

**Comment by TylerLeonhardt** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5428870&bid=17802194)

File: src/PowerShellEditorServices/Services/PowerShellContext/Session/Host/TerminalPSHostUserInterface.cs [hunk 2]
```diff
@@ -70,6 +72,36 @@ internal class TerminalPSHostUserInterface : EditorServicesPSHostUserInterface
...
+        internal override ConsoleColor ProgressForegroundColor
+        {
+            get
+            {
+                return (ConsoleColor) _internalHostPrivateData.Properties["ProgressForegroundColor"].Value;
+            }
+            set
+            {
+                _internalHostPrivateData.Properties["ProgressForegroundColor"].Value = value;
+            }
```
Comment (SeeminglyScience) [inline src/PowerShellEditorServices/Services/PowerShellContext/Session/Host/TerminalPSHostUserInterface.cs:87]: Optional nit

```suggestion
            get => (ConsoleColor)_privateData.Properties["ProgressForegroundColor"].Value;
            set => _privateData.Properties["ProgressForegroundColor"].Value = value;
```

File: src/PowerShellEditorServices/Services/PowerShellContext/Session/Host/TerminalPSHostUserInterface.cs [hunk 3]
```diff
@@ -70,6 +72,36 @@ internal class TerminalPSHostUserInterface : EditorServicesPSHostUserInterface
...
+        internal override ConsoleColor ProgressBackgroundColor
+        {
+            get
+            {
+                return (ConsoleColor) _internalHostPrivateData.Properties["ProgressBackgroundColor"].Value;
+            }
+            set
+            {
+                _internalHostPrivateData.Properties["ProgressBackgroundColor"].Value = value;
+            }
```
Comment (SeeminglyScience) [inline src/PowerShellEditorServices/Services/PowerShellContext/Session/Host/TerminalPSHostUserInterface.cs:102]: Optional nit:

```suggestion
            get => (ConsoleColor)_internalHostPrivateData.Properties["ProgressBackgroundColor"].Value;
            set => _internalHostPrivateData.Properties["ProgressBackgroundColor"].Value = value;
```

File: src/PowerShellEditorServices/Services/PowerShellContext/Session/Host/TerminalPSHostUserInterface.cs (hunk 1 above)
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices/Services/PowerShellContext/Session/Host/TerminalPSHostUserInterface.cs:26]:
```suggestion
        private readonly ConsoleReadLine _consoleReadLine;
```

**Comment by TylerLeonhardt** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5428870&bid=17802194)

**Comment by TylerLeonhardt** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5428870&bid=17802194)

**Comment by TylerLeonhardt** [comment]
why are these breaks happening? There are no references of `internalHostUI` in that file......

**Comment by TylerLeonhardt** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5428870&bid=17802194)

**Pull request merged**